src/Board.hs

```diff
@@ -4,7 +4,9 @@ module Board (
     isStartingLayout,
     Dimensions(..),
     Board(..),
-    SlideError(..)
+    SlideError(..),
+    Width,
+    Height
```

Comment:
In my opinion, `type` here is fine.

However, if you want the compiler to be picky, you can do
```
newtype Width = Width { getWidth :: Int } 
newtype Height = Height { getHeight :: Int }
```
or
```
data Width = Width Int
data Height = Height Int
```
.

The difference here is that `data` actually makes a "struct" with a pointer to the int, while newtype has absolutely no runtime overhead --- the compiled code behaves as though you were just using an Int. I'm not sure, however, if `data Width = Width {-# UNPACK #-} !Int` performs the same as `newtype`.